pkg/frontend/nvme.go

```diff
@@ -21,15 +21,22 @@ import (
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
+// TODO: consider using https://pkg.go.dev/net#TCPAddr
 type tcpSubsystemListener struct {
 	listenAddr net.IP
+	listenPort string
 }
 
 // NewTCPSubsystemListener creates a new instance of tcpSubsystemListener
 func NewTCPSubsystemListener(listenAddr string) SubsystemListener {
-	parsedAddr := net.ParseIP(listenAddr)
+	host, port, err := net.SplitHostPort(listenAddr)
+	if err != nil {
+		log.Panicf("Invalid ip:port tuple: %v", listenAddr)
+	}
+
+	parsedAddr := net.ParseIP(host)
 	if parsedAddr == nil {
-		log.Panicf("Invalid ip address: %v", listenAddr)
+		log.Panicf("Invalid ip address: %v", host)
 	}
 
 	if parsedAddr.To4() == nil {
```

Comment:
one potential improvement for future: we can determine protocol ipv4 or ipv6 by calling `parsedAddr.To4()` and `parsedAddr.To6()` and pass the required protocol into listener parameters